packages/terra-date-time-picker/tests/wdio/date-time-picker-spec.js

```diff
@@ -647,3 +647,791 @@ Terra.describeViewports('DateTimePicker', ['tiny', 'large'], () => {
+          $('input[name="terra-date-year-input"]').click();
+          browser.keys('2016');
+          $('input[name="terra-date-month-input"]').click();
+          browser.keys('05');
+          $('input[name="terra-time-second-input"]').click();
+          browser.keys('a');
+
+          const today = moment.utc();
+          expect($('input[name="terra-date-year-input"]')).toHaveValue(today.year().toString());
+          expect($('input[name="terra-date-month-input"]')).toHaveValue((`0${(today.month() + 1)}`).slice(-2));
+          expect($('input[name="terra-date-day-input"]')).toHaveValue((`0${today.date()}`).slice(-2));
+          expect($('input[name="terra-time-hour-input"]')).toHaveValue('12');
+          expect($('input[name="terra-time-minute-input"]')).toHaveValue('00');
+          expect($('input[name="terra-time-second-input"]')).toHaveValue('00');
+          expect($('button[class*="meridiem"]')).toHaveAttrContaining('class', 'is-selected');
+        });
+      });
+
+      describe('if valid date', () => {
+        it('in the hour input should not change date and should leave it as AM', () => {
+          browser.url('/raw/tests/terra-date-time-picker/date-time-picker/date-time-picker-default-seconds-twelve-hour');
+          browser.refresh();
+          $('input[name="terra-time-hour-input"]').click();
+          browser.keys('a');
+
+          expect($('input[name="terra-date-year-input"]')).toHaveValue('2017');
+          expect($('input[name="terra-date-month-input"]')).toHaveValue('08');
+          expect($('input[name="terra-date-day-input"]')).toHaveValue('08');
+          expect($('input[name="terra-time-hour-input"]')).toHaveValue('10');
+          expect($('input[name="terra-time-minute-input"]')).toHaveValue('30');
+          expect($('input[name="terra-time-second-input"]')).toHaveValue('00');
+          expect($('button[class*="meridiem"]')).toHaveAttrContaining('class', 'is-selected');
+        });
+
+        it('in the minute input should not change date and should leave it as AM', () => {
+          browser.url('/raw/tests/terra-date-time-picker/date-time-picker/date-time-picker-default-seconds-twelve-hour');
+          browser.refresh();
+          $('input[name="terra-time-minute-input"]').click();
+          browser.keys('a');
+
+          expect($('input[name="terra-date-year-input"]')).toHaveValue('2017');
+          expect($('input[name="terra-date-month-input"]')).toHaveValue('08');
+          expect($('input[name="terra-date-day-input"]')).toHaveValue('08');
+          expect($('input[name="terra-time-hour-input"]')).toHaveValue('10');
+          expect($('input[name="terra-time-minute-input"]')).toHaveValue('30');
+          expect($('input[name="terra-time-second-input"]')).toHaveValue('00');
+          expect($('button[class*="meridiem"]')).toHaveAttrContaining('class', 'is-selected');
+        });
+
+        it('in the second input should not change date and should leave it as AM', () => {
```

Comment:
A couple test scenarios I'm not seeing, but think are important:
* 12-hour PM time, press A and see the time switch to PM
* 24-hour clock, press A and see nothing changes.